sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/listeners/OnMultipleRequestListener.java

```diff
@@ -21,10 +21,22 @@
 	public OnMultipleRequestListener(){
 		setListenerType(UPDATE_LISTENER_TYPE_MULTIPLE_REQUESTS);
 		correlationIds = new Vector<>();
+		final OnMultipleRequestListener reference = this;
 
 		rpcResponseListener = new OnRPCResponseListener() {
 			@Override
 			public void onResponse(int correlationId, RPCResponse response) {
+				update(correlationId);
+			}
+
+			@Override
+			public void onError(int correlationId, Result resultCode, String info) {
+				super.onError(correlationId, resultCode, info);
+				reference.onError(correlationId, resultCode, info);
+				update(correlationId);
+			}
+
+			private void update(int correlationId){
 				correlationIds.remove(Integer.valueOf(correlationId));
```

Comment:
Why not just change the vector to `int` instead of `Integer`?

Reply:
Vector can't use primitive type